gnucash/register/ledger-core/gncEntryLedgerModel.c

```diff
@@ -576,9 +575,7 @@ static char * get_date_help (VirtualLocation virt_loc, gpointer user_data)
 
     gnc_date_cell_get_date ((DateCell *) cell, &cell_time, FALSE);
 
-    date_string = gnc_print_time64 (cell_time, _("%A %d %B %Y"));
-
-    return g_strdup (date_string);
+    return gnc_print_time64 (cell_time, _("%A %d %B %Y"));
```

Comment:
I tried the branch changed to return a pointer to a `static gchar*` however it segfaults, because the `get_date_help` eventually is assigned to https://github.com/Gnucash/gnucash/blob/b811bc6c8b87e29c41334a6a5c53aefec1059044/gnucash/gnome/gnc-plugin-page-register.c#L5268 which is freed appropriately via https://github.com/Gnucash/gnucash/blob/b811bc6c8b87e29c41334a6a5c53aefec1059044/gnucash/gnome/gnc-plugin-page-register.c#L5270. So, returning `gnc_print_time64(...)` is the appropriate approach here.

Reply:
Attached: back trace when returning a static string:

```
#0  0x00007ffff7b4fa84 in g_logv () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#1  0x00007ffff7b4fd13 in g_log () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff587ffb7 in gnc_split_register_get_date_help (virt_loc=..., user_data=0x555555f8f320)
    at /home/chris/sources/gnucash/maint/gnucash/register/ledger-core/split-register-model.c:1011
#3  0x00007ffff5847fda in gnc_table_get_help (table=0x5555560323d0) at /home/chris/sources/gnucash/maint/gnucash/register/register-core/table-allgui.c:454
#4  0x00007ffff7a6b647 in gnc_plugin_page_help_changed_cb (gsr=0x555556054180, register_page=0x555556017c10)
    at /home/chris/sources/gnucash/maint/gnucash/gnome/gnc-plugin-page-register.c:5268
#5  0x00007ffff6df0bcf in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff6e1a81b in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#7  0x00007ffff6e0e638 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00007ffff6e0eaa9 in g_signal_emit_by_name () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#9  0x00007ffff7a89bf0 in gsr_emit_simple_signal (gsr=0x555556054180, sigName=0x7ffff7ac359c "help-changed")
    at /home/chris/sources/gnucash/maint/gnucash/gnome/gnc-split-reg.c:2593
#10 0x00007ffff7a89b5a in gsr_emit_help_changed (reg=0x555556057140, user_data=0x555556054180) at /home/chris/sources/gnucash/maint/gnucash/gnome/gnc-split-reg.c:2572
#11 0x00007ffff6df0bcf in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#12 0x00007ffff6e1a81b in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007ffff6e0e638 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#14 0x00007ffff6e0eaa9 in g_signal_emit_by_name () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#15 0x00007ffff794e7e6 in gnucash_sheet_redraw_help (sheet=0x555555c9d650) at /home/chris/sources/gnucash/maint/gnucash/register/register-gnome/gnucash-sheet.c:763
#16 0x00007ffff79560d2 in table_ui_redraw_cb (table=0x5555560323d0) at /home/chris/sources/gnucash/maint/gnucash/register/register-gnome/table-gnome.c:127
#17 0x00007ffff58494b5 in gnc_table_enter_update (table=0x5555560323d0, virt_loc=..., cursor_position=0x7fffffffc72c, start_selection=0x7fffffffc730, end_selection=0x7fffffffc734)
    at /home/chris/sources/gnucash/maint/gnucash/register/register-core/table-allgui.c:1123
#18 0x00007ffff794d5e5 in gnucash_sheet_activate_cursor_cell (sheet=0x555555c9d650, changed_cells=1)
    at /home/chris/sources/gnucash/maint/gnucash/register/register-gnome/gnucash-sheet.c:396
#19 0x00007ffff7952c97 in gnucash_sheet_table_load (sheet=0x555555c9d650, do_scroll=1) at /home/chris/sources/gnucash/maint/gnucash/register/register-gnome/gnucash-sheet.c:2246
#20 0x00007ffff7956258 in gnc_table_refresh_gui (table=0x5555560323d0, do_scroll=1) at /home/chris/sources/gnucash/maint/gnucash/register/register-gnome/table-gnome.c:179
#21 0x00007ffff587dfea in gnc_split_register_load (reg=0x555555f8f320, slist=0x55555594bf40 = {...}, default_account=0x555555db2fa0)
    at /home/chris/sources/gnucash/maint/gnucash/register/ledger-core/split-register-load.c:800
#22 0x00007ffff586a468 in gnc_ledger_display_refresh_internal (ld=0x555556003130, splits=0x55555594bf40 = {...})
    at /home/chris/sources/gnucash/maint/gnucash/register/ledger-core/gnc-ledger-display.c:874
#23 0x00007ffff586a5c9 in gnc_ledger_display_refresh (ld=0x555556003130) at /home/chris/sources/gnucash/maint/gnucash/register/ledger-core/gnc-ledger-display.c:897
```

Reply:
So it is. I was mislead by the deleted `const char*` local.